### PR TITLE
adds log levels and log messages

### PR DESCRIPTION
**Pull request by arolonss**

### CODE REVIEW

**Comment by hazelbakerharvey** [comment]
Did you add an error level logging?

File: users-api/src/main/java/com/ss/eastcoderbank/usersapi/service/AuthService.java [hunk 1]
```diff
@@ -26,14 +33,21 @@
...
     public void userLogin(LoginDto loginDto) throws Exception {
+        LOGGER.trace("AuthService.userLogin reached...");
+        LOGGER.info("Mapping loginDto to user entity...");
         User user = loginMapper.mapToEntity(loginDto);
+        LOGGER.debug(String.valueOf(user));
     }
 
     public UserPrincipal findUser(String s) throws UsernameNotFoundException {
+        LOGGER.trace("AuthService.findUser reached...");
+        LOGGER.debug(s);
```
Comment (hazelbakerharvey) [inline users-api/src/main/java/com/ss/eastcoderbank/usersapi/service/AuthService.java]: Just commenting on this example, i think debug should be used for only debugging purposes. not necessarily in any develop version

File: users-api/src/main/java/com/ss/eastcoderbank/usersapi/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -46,15 +51,16 @@ public JwtAuthenticationFilter(AuthenticationManager authenticationManager, @Val
...
             credentials = new ObjectMapper().readValue(request.getInputStream(), LoginDto.class);
         } catch (IOException e) {
-            logger.error(e.getMessage());
+            LOGGER.error(e.getMessage());
             throw new RuntimeException(e);
-            //e.printStackTrace();
+
         } catch (NullPointerException e) {
-            logger.error(e.getMessage());
+            LOGGER.error(e.getMessage());
```
Comment (hazelbakerharvey) [inline users-api/src/main/java/com/ss/eastcoderbank/usersapi/security/JwtAuthenticationFilter.java]: Nvm about my previous question about error level! i was just looking at the properties. Why use get message here and other places a custom message?

File: users-api/src/main/java/com/ss/eastcoderbank/usersapi/security/JwtAuthorizationFilter.java [hunk 3]
```diff
@@ -39,6 +44,9 @@ public JwtAuthorizationFilter(AuthenticationManager authenticationManager, UserR
 
     @Override
     protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain chain) throws ServletException, IOException {
+        LOGGER.trace("JwtAuthorizationFilter.doFilterInternal reached...");
```
Comment (hazelbakerharvey) [inline users-api/src/main/java/com/ss/eastcoderbank/usersapi/security/JwtAuthorizationFilter.java:47]: I think trace is also used for debugging, but for more detail. https://www.papertrail.com/solution/tips/logging-in-java-best-practices-and-tips/

File: users-api/src/main/java/com/ss/eastcoderbank/usersapi/service/UserService.java [hunk 4]
```diff
@@ -50,11 +55,18 @@
     private CreateUserMapper createUserMapper;
 
     public Page<UserDto> getUsersByRole(String title, Pageable page) {
+        LOGGER.trace("UserService.getUsersByRole reached...");
```
Comment (hazelbakerharvey) [inline users-api/src/main/java/com/ss/eastcoderbank/usersapi/service/UserService.java]: Like this trace and debug i think would be better as info

File: users-api/src/main/java/com/ss/eastcoderbank/usersapi/controller/UserController.java [hunk 5]
```diff
@@ -25,37 +29,75 @@
...
+     * @param pageNumber
+     * @param pageSize
+     * @return a page of inactive users
+     */
     @PreAuthorize("hasAuthority('Admin')")
     @GetMapping("/users/inactive")
     public Page<UserDto> getInactiveUsers(@RequestParam(name="page") Integer pageNumber, @RequestParam(name="size") Integer pageSize) {
+        LOGGER.trace("UserController.getInactiveUsers");
+        LOGGER.debug("page: "+ pageNumber + " size: "+ pageSize);
+        LOGGER.info("Getting inactive users...");
```
Comment (asernass) [inline users-api/src/main/java/com/ss/eastcoderbank/usersapi/controller/UserController.java]: I wouldn't put trace and debug logs in the controller handling the request. I would move those to the services that call them.

File: users-api/src/main/java/com/ss/eastcoderbank/usersapi/service/UserService.java [hunk 6]
```diff
@@ -50,11 +55,18 @@
...
+        LOGGER.debug("role title: "+ title);
+        LOGGER.info("Getting users by role page from User Repository...");
         return userRepository.findUserByRoleTitle(title, page).map(user -> userMapper.mapToDto(user));
 
     }
 
     public Integer createUser(CreateUserDto createUserDto) throws DuplicateConstraintsException {
+        LOGGER.trace("UserService.createUser reached...");
+        LOGGER.debug("DTO: "+ createUserDto);
+        LOGGER.info("Creating user profile...");
```
Comment (hazelbakerharvey) [inline users-api/src/main/java/com/ss/eastcoderbank/usersapi/service/UserService.java:64]: Same here. this pattern of trace, debug and info should probably be all info. I think also we might not need this level of detail like logging the dto itself

File: users-api/src/main/java/com/ss/eastcoderbank/usersapi/service/UserService.java [hunk 7]
```diff
@@ -71,14 +83,16 @@ public Integer createUser(CreateUserDto createUserDto) throws DuplicateConstrain
             if (t instanceof ConstraintViolationException) {
                 handleUniqueConstraints(((ConstraintViolationException) t).getConstraintName());
             }
-            log.info(e.getMessage(), e);
+            LOGGER.info(e.getMessage(), e);
```
Comment (hazelbakerharvey) [inline users-api/src/main/java/com/ss/eastcoderbank/usersapi/service/UserService.java:82]: probably should be error

File: users-api/src/main/java/com/ss/eastcoderbank/usersapi/service/UserService.java [hunk 8]
```diff
@@ -92,24 +106,27 @@ public Integer updateUser(UpdateProfileDto updateProfileDto, Integer id) {
             if (thr instanceof ConstraintViolationException) {
                 handleUniqueConstraints(((ConstraintViolationException) thr).getConstraintName());
```
Comment (hazelbakerharvey) [inline users-api/src/main/java/com/ss/eastcoderbank/usersapi/service/UserService.java:101]: needs an error log?

File: users-api/src/main/java/com/ss/eastcoderbank/usersapi/service/UserService.java [hunk 9]
```diff
@@ -92,24 +106,27 @@ public Integer updateUser(UpdateProfileDto updateProfileDto, Integer id) {
...
      *@param asc the sort direction
      *@param sort the property to be sorted by
      */
     public Page<UserDto> getUsers(Integer pageNumber, Integer pageSize, boolean asc, String sort) {
+        LOGGER.trace("UserService.getUsers");
+        LOGGER.debug(" page number: "+ pageNumber +" page size: "+ pageSize + " asc?: "+ asc + " sorted by: "+ sort);
+        LOGGER.info("Preparing page request to send to userRepository...");
         Page<UserDto> req;
         if (sort != null) {
             req = userRepository.findAll(PageRequest.of(pageNumber, pageSize, Sort.by(asc ? Sort.Direction.ASC : Sort.Direction.DESC, sort))).map(user -> userMapper.mapToDto(user));
```
Comment (hazelbakerharvey) [inline users-api/src/main/java/com/ss/eastcoderbank/usersapi/service/UserService.java:117]: could be even more verbose, ie "Sorting enabled"

File: users-api/src/main/java/com/ss/eastcoderbank/usersapi/service/UserService.java [hunk 10]
```diff
@@ -119,29 +136,40 @@ public Integer updateUser(UpdateProfileDto updateProfileDto, Integer id) {
...
     }
 
     public List<UserRole> getRoles() {
+        LOGGER.trace("UserController.deactivateUser");
+        LOGGER.info("Gettting roles from userRoleRepository.");
         return userRoleRepository.findAll();
     }
 
 
     protected void handleUniqueConstraints(String constraint) {
```
Comment (hazelbakerharvey) [inline users-api/src/main/java/com/ss/eastcoderbank/usersapi/service/UserService.java]: this needs a bunch more error logs?

**Review by hazelbakerharvey — CHANGES_REQUESTED** [review]
I left a bunch of comments! Generally I think this could use more error logging esp with custom constraints etc. I also think we dont need trace, debug and info all together. I would remove the trace and debug mostly, but if you want that info switch them to info logs. I like that you have made the amount of logs diff depending on the environment!!

File: users-api/src/main/java/com/ss/eastcoderbank/usersapi/security/JwtAuthenticationFilter.java (hunk 2 above)
Comment (arolonss) [reply users-api/src/main/java/com/ss/eastcoderbank/usersapi/security/JwtAuthenticationFilter.java]: I left was not sure if it was serving a purpose to anyone else at the time but I am happy to change it to a custom message if needed.

**Comment by arolonss** [comment]
> I left a bunch of comments! Generally I think this could use more error logging esp with custom constraints etc. I also think we dont need trace, debug and info all together. I would remove the trace and debug mostly, but if you want that info switch them to info logs. I like that you have made the amount of logs diff depending on the environment!!

Great! Thanks for the review, I will get to makes those changes today.

File: users-api/src/main/java/com/ss/eastcoderbank/usersapi/controller/UserController.java (hunk 5 above)
Comment (arolonss) [reply users-api/src/main/java/com/ss/eastcoderbank/usersapi/controller/UserController.java]: > I wouldn't put trace and debug logs in the controller handling the request. I would move those to the services that call them.

Thanks, I will get to that today.

**Pull request merged**